api/src/main/java/org/openmrs/module/reporting/cohort/definition/library/BuiltInCohortDefinitionLibrary.java

```diff
@@ -150,4 +152,23 @@ public CohortDefinition getPersonWithAttribute() {
         cd.addParameter(new Parameter("values", "reporting.parameter.values", String.class, List.class, null));
         return cd;
     }
+
+    @DocumentedDefinition("programEnrollment")
+    public CohortDefinition getProramEnrollment(){
+        ProgramEnrollmentCohortDefinition cd = new ProgramEnrollmentCohortDefinition();
+        cd.addParameter(new Parameter("programs", "reporting.parameters.programs", Program.class, List.class, null));
+        cd.addParameter(new Parameter("enrolledOnOrAfter", "reporting.parameters.startDate", Date.class));
+        cd.addParameter(new Parameter("enrolledOnOrBefore", "reporting.parameters.endDate", Date.class));
+        cd.addParameter(new Parameter("completedOnOrAfter", "reporting.parameters.startDate", Date.class));
+        cd.addParameter(new Parameter("completedOnOrBefore", "reporting.parameters.endDate", Date.class));
+        cd.addParameter(new Parameter("locationList", "reporting.parameters.programs", Location.class, List.class, null));
+        return new MappedParametersCohortDefinition(cd,
+            "programs", "programs",
+            "enrolledOnOrAfter", "enrolledStartDate",
+            "enrolledOnOrBefore", "enrolledEndDate",
+            "completedOnOrAfter", "compeletedStartDate",
+            "completedOnOrBefore", "completedEndDate",
+            "locationList", "locations"
```

Comment:
I see no good reason to change the names of these date parameters.  I would just leave them as "enrolledOnOrAfter", etc.  Much more clear to me.  If we had thought that "enrolledStartDate" was clearer, we would have named them that way in the definition originally :)